metaflow/plugins/list_cli.py

```diff
@@ -0,0 +1,101 @@
+from metaflow._vendor import click
+from metaflow import Flow, Run
+from metaflow import util
+from metaflow.exception import MetaflowNotFound, CommandException
+import json
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(help="List objects pertaining to your flow.")
+def list():
+    pass
+
+
+def _fetch_runs(flow_name, num_runs, namespace):
+    counter = 1
+    try:
+        flow = Flow(flow_name)
+    except MetaflowNotFound:
+        flow = None
+    run_list = []
+    if flow:
+        for run in Flow(flow_name).runs():
+            if counter > num_runs:
+                break
+            tags = [t for t in run.tags]
+            if namespace is None or namespace in tags:
+                counter += 1
+                run_list.append(
+                    dict(
+                        created=str(run.created_at),
+                        name=flow_name,
+                        id=run.id,
+                        status=run.successful,
+                        finished=run.finished,
+                        tags=tags,
+                    )
+                )
+    return run_list
+
+
+@click.option(
+    "--num-runs",
+    default=10,
+    type=click.IntRange(1, None),
+    help="Number of runs to show.",
+)
+@click.option(
+    "--as-json",
+    default=False,
+    is_flag=True,
+    help="Print run list as a JSON object",
+)
+@click.option(
+    "--file",
+    default=None,
+    help="Save the run list to file as json.",
+)
+@click.option("--user", default=None, help="List runs for the given user.")
+@click.option("--namespace", default=None, help="List runs for the given namespace.")
+@click.option(
+    "--my-runs",
+    default=False,
+    is_flag=True,
+    help="List all my runs.",
+)
+@list.command(help="List recent runs for your flow.")
+@click.pass_context
+def runs(ctx, num_runs, as_json, file, user, my_runs, namespace):
+    if user and my_runs:
+        raise CommandException("--user and --my-runs are mutually exclusive.")
+    if user and namespace:
+        raise CommandException("--user and --namespace are mutually exclusive.")
+    if my_runs and namespace:
+        raise CommandException("--my-runs and --namespace are mutually exclusive.")
+
+    if my_runs:
+        namespace = "user:{}".format(util.get_username())
+    if user:
+        namespace = "user:{}".format(user)
```

Comment:
default should probably be my-runs (which is what it was before).

Reply:
@romain-intel Just to clarify, I think what you are saying is that because of the call to `namespace(namespace)` in `cli.py` makes the `--my-runs` flags necessary?  Is my understanding correct? 

cc: @savingoyal incase I misunderstood your instructions earlier to add this flag as well

Reply:
no, I meant that right now, if you don't specify `my-runs` or `user`, it will do `namespace(None)` (or at least you should do that -- see my comment below) which means "no namespace filter". I am saying that if there is nothing specified, we should default to the behavior of `my-runs` because that is probably the most natural thing. We can still have an option (to be explicit). Of course that does bring the "how do you list all runs across namespaces" but I still feel that having the `my-runs` be the "default" behavior is probably more natural. At least it is the one that best answers the user story that @valayDave had in the original issue.

Reply:
ok I resolved this [with these changes](https://github.com/Netflix/metaflow/pull/1037#issuecomment-1123825173)